lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,216 @@
+use bytes::Bytes;
+use chrono::{DateTime, NaiveDateTime, Utc};
+use lookup::path;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use smallvec::{smallvec, SmallVec};
+use std::collections::HashMap;
+use value::Kind;
+use vector_core::{
+    config::{log_schema, DataType},
+    event::Event,
+    schema,
+};
+
+use super::Deserializer;
+
+/// GELF Message fields. Definitions from https://docs.graylog.org/docs/gelf
+pub mod gelf_fields {
+
+    /// <not a field> The latest version of the GELF specificaiton.
+    pub const GELF_VERSION: &str = "1.1";
+
+    /// (required) GELF spec version – “1.1”.
+    pub const VERSION: &str = "version";
+
+    /// (required) The name of the host, source or application that sent this message.
+    pub const HOST: &str = "host";
+
+    /// (required) A short descriptive message.
+    pub const SHORT_MESSAGE: &str = "short_message";
+
+    /// (optional) A long message that can i.e. contain a backtrace
+    pub const FULL_MESSAGE: &str = "full_message";
+
+    /// (optional) Seconds since UNIX epoch with optional decimal places for milliseconds.
+    ///  SHOULD be set by client library. Will be set to the current timestamp (now) by the server if absent.
+    pub const TIMESTAMP: &str = "timestamp";
+
+    /// (optional) The level equal to the standard syslog levels. default is 1 (ALERT).
+    pub const LEVEL: &str = "level";
+
+    /// (optional) (deprecated) Send as additional field instead.
+    pub const FACILITY: &str = "facility";
+
+    /// (optional) (deprecated) The line in a file that caused the error (decimal). Send as additional field instead.
+    pub const LINE: &str = "line";
+
+    /// (optional) (deprecated) The file (with path if you want) that caused the error. Send as additional field instead.
+    pub const FILE: &str = "file";
+}
+pub use gelf_fields::*;
+
+/// Config used to build a `GelfDeserializer`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct GelfDeserializerConfig;
+
+impl GelfDeserializerConfig {
+    /// Build the `GelfDeserializer` from this configuration.
+    pub fn build(&self) -> GelfDeserializer {
+        GelfDeserializer::default()
+    }
+
+    /// Return the type of event built by this deserializer.
+    pub fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    /// The schema produced by the deserializer.
+    pub fn schema_definition(&self) -> schema::Definition {
+        schema::Definition::empty()
+            .with_field(VERSION, Kind::bytes(), None)
+            .with_field(HOST, Kind::bytes(), None)
+            .with_field(SHORT_MESSAGE, Kind::bytes(), None)
+            .optional_field(FULL_MESSAGE, Kind::bytes(), None)
+            .optional_field(TIMESTAMP, Kind::timestamp(), None)
+            .optional_field(LEVEL, Kind::integer(), None)
+            .optional_field(FACILITY, Kind::bytes(), None)
+            .optional_field(LINE, Kind::integer(), None)
```

Comment:
Wouldn't this be a float based on the `(decimal)` comment above, or am I misremembering our types.

Reply:
Mmm , good question. I'll see if I can try out the open source graylog and see if I can get to the bottom of that. Seems weird that a line number in a file would be a non-integer? But that `(decimal)` does suggest.